---
title: C28105
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: c0906c1e876a13049dfe1d1e7727c22528f8e8ed
ms.sourcegitcommit: 21d667104199c2493accec20c2388cf674b195c3
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 02/08/2019
ms.locfileid: "55946089"
---
# <a name="c28105"></a>C28105
警告 C28105:流失資源，因為發生例外狀況

 引發例外狀況時，不會釋放指定的資源。 路徑所指定的陳述式可以引發例外狀況。 這項警告是類似於警告[C28103](../code-quality/c28103.md)，不過在此情況下的相關例外狀況。

## <a name="example"></a>範例
 下列程式碼範例會產生此警告：

```
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 下列程式碼範例可避免此警告：

```
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```